File: fixtures/scale/scala/medium/src/calc/ast/VariableExpr.scala
package calc.ast

final case class VariableExpr(name: String) extends Expr {

  override def describe: String = name

  def isConstantName: Boolean =
    VariableExpr.ConstantNames.contains(name)
}

object VariableExpr {
  private val ConstantNames = Set("pi", "e", "phi", "tau")
}

File: fixtures/scale/kotlin/medium/src/calc/ast/VariableExpr.kt
package calc.ast

data class VariableExpr(val name: String) : Expr {

    override fun describe(): String = name

    fun isConstantName(): Boolean = name in CONSTANT_NAMES

    companion object {
        private val CONSTANT_NAMES = setOf("pi", "e", "phi", "tau")
    }
}
